### PR TITLE
[FEAT] add showError to preprocess options

### LINKED ISSUE #3: [FEAT] Add ability to turn off diagnostics/errors
**Describe the solution you'd like**
I would like to turn on/off an option that would toggle the diagnostics/errors in the console.
i.e. disable this:

https://github.com/pyoner/svelte-typescript/blob/ab679023478ee4065bf627a0d49d63bb8956275c/packages/preprocess/src/svelte-ts-preprocess.ts#L192-L200

**Additional context**
I currently use the svelte plugin for VSCode that supports ts inside script elements so I don't need diagnostics/errors in the console.

I mainly want an option to do this because the preprocessor shows false positives in the console. This is because I use rollup plugins for other TS files and the file gets transformed before it gets passed to the typescript rollup plugin.

To get this functionality asap in my project I just copied `svelte-ts-preprocess` from dist and commented out the mentioned lines.
I can put up a PR for this if you want.

### PR DESCRIPTION
**Pull request by Teoxoy**
closes #3

### CODE REVIEW

**Comment by pyoner** [comment]
@Teoxoy thank you for the PR!
Can you change the option name `showErrors` to `hideErrors`?
Because errors are enabled by default.

**Comment by Teoxoy** [comment]
They are on by default `showErrors: opts.showErrors ? opts.showErrors : true`

Alltho I overlooked something, even if you pass false they will show. I need to change that check to check if `showErrors !== undefined`.

Or do you still want me to rename the option to hideErrors?

**Comment by pyoner** [comment]
I mean to change not only the name, but also the logic
```typescript
// by default errors are enabled
let opts = {}
if (!opts.hideErrors) {
  // show errors
}
```

**Comment by Teoxoy** [comment]
Done

File: packages/preprocess/src/svelte-ts-preprocess.ts [hunk 1]
```diff
@@ -142,7 +142,7 @@ export function readConfigFile(env: Env, path?: string) {
 export interface PreprocessOptions {
   compilerOptions: ts.CompilerOptions
   env: Env
-  showErrors: boolean
+  hideErrors: boolean
```
Comment (pyoner) [inline packages/preprocess/src/svelte-ts-preprocess.ts]: Can you make `hideErrors` as optional property?
```typescript
export interface PreprocessOptions {
  compilerOptions: ts.CompilerOptions
  env: Env
  hideErrors?: boolean
}
```

File: packages/preprocess/src/svelte-ts-preprocess.ts [hunk 2]
```diff
@@ -151,7 +151,7 @@ export function createPreprocessOptions(opts?: Partial<PreprocessOptions>): Prep
   return {
     compilerOptions: opts.compilerOptions ? opts.compilerOptions : defaultCompilerOptions,
     env: opts.env ? opts.env : createEnv(),
-    showErrors: opts.showErrors ? opts.showErrors : true
+    hideErrors: !!opts.hideErrors
```
Comment (pyoner) [inline packages/preprocess/src/svelte-ts-preprocess.ts]: I think `hideErrors` no need here

Comment (Teoxoy) [reply packages/preprocess/src/svelte-ts-preprocess.ts]: It has to be passed down trough that function doesn't it?

File: packages/preprocess/src/svelte-ts-preprocess.ts (hunk 1 above)
Comment (Teoxoy) [reply packages/preprocess/src/svelte-ts-preprocess.ts]: It shouldn't be optional there.
The ops object is already `Partial<PreprocessOptions>`

Comment (pyoner) [reply packages/preprocess/src/svelte-ts-preprocess.ts]: You are right, sorry

Comment (Teoxoy) [reply packages/preprocess/src/svelte-ts-preprocess.ts]: np :)

**Comment by pyoner** [comment]
@Teoxoy 
I published https://www.npmjs.com/package/@pyoner/svelte-ts-preprocess/v/1.1.0

**Comment by Teoxoy** [comment]
Thanks!

**Pull request merged**